chips/stm32f4xx/src/clocks/clocks.rs

```diff
@@ -431,8 +439,13 @@ impl<'a, ChipSpecs: ChipSpecsTrait> Clocks<'a, ChipSpecs> {
     /// + [Err]\([ErrorCode::FAIL]\) if the source apart from HSI is already enabled.
     pub fn set_mco1_clock_source(&self, source: MCO1Source) -> Result<(), ErrorCode> {
         match source {
+            MCO1Source::HSE => {
+                if !self.hse.is_enabled() {
+                    return Err(ErrorCode::FAIL);
+                }
+            }
             MCO1Source::PLL => {
-                if self.pll.is_enabled() {
+                if !self.pll.is_enabled() {
```

Comment:
From documentation, page 166: `Clock source selection may generate glitches on MCO1. It is
highly recommended to configure these bits only after reset before enabling the external
oscillators and PLL.` I decided to follow the recommandation and allow setting MCO1 clock source before enabling PLL clock => PPL must be disabled before setting it as the MCO1 source.

Reply:
Thank you. I missed this detail. Change removed